### PR TITLE
add help

### PR DESCRIPTION
**Pull request by alex-courtis**
This is a skeleton / draft... it needs some work from you.

### CODE REVIEW

File: doc/nvim-tree.lua-float-preview.txt [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      up = { "<C-e>", "<C-u>" },
+    },
+  })
+<
+*float-preview.scroll_lines*
+Maximum number of lines the preview window will scroll vertically.
+Type: `number`, Default: `20`
+
+*float-preview.mapping*
+Float window mappings.
```
Comment (JMarkin) [inline doc/nvim-tree.lua-float-preview.txt]: I want this mapping used for display in float window if you focus on nvim.tree buffer
For example 
- down - scroll down by text
- up - scroll up by text

File: doc/nvim-tree.lua-float-preview.txt [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+|nvim-tree| buffer mappings are needed to activate functionality. These take
+the form of a wrapper around |nvim-tree-api| calls in your |nvim-tree.on_attach|
+function. See |nvim-tree-mappings| for more details.
+
+Common functions to wrap may include:
+- |nvim-tree-api.node| for opening and previewing nodes
+- |nvim-tree-api.tree| to handle tree open/close
+- |nvim-tree-api.fs| for file creation
+
+A new???? (JMarkin TODO) instance of float preview must? be created: >
```
Comment (JMarkin) [inline doc/nvim-tree.lua-float-preview.txt]: It depends on user config. For example I used lazy.nvim and I want that this plugin will load after nvim-tree. Example In readme I show more correctable usage by my opinion.

`:new` creates instance of float window and user can work with instance but it needs to attach to nvim-tree buffer.

After I wrote message for you I thought may be without `new` and to create public method `attach` and `float_close_wrap`, what do you think?

File: doc/nvim-tree.lua-float-preview.txt [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+    Return: ~
+      (boolean) buffer is a floating preview
+
+TODO JMarkin I am not sure what else you want to make public. :new ?
+
+==============================================================================
+ 5. DIAGNOSTICS                                  *float-preview-diagnostics*
+
+TODO JMarkin I am not explaining why this is necessary
```
Comment (JMarkin) [inline doc/nvim-tree.lua-float-preview.txt]: In my usages if I enabled nullls or other external diagnostic plugin, it attached to buffer automatically and it loads system. I think many users want to recognize what it is float buffer from nvim-tree

File: doc/nvim-tree.lua-float-preview.txt (hunk 2 above)
Comment (alex-courtis) [reply doc/nvim-tree.lua-float-preview.txt]: > After I wrote message for you I thought may be without new and to create public method attach and float_close_wrap, what do you think?

That is sensible. It follows the pattern of most nvim plugins - a require'd singleton.

File: doc/nvim-tree.lua-float-preview.txt (hunk 1 above)
Comment (alex-courtis) [reply doc/nvim-tree.lua-float-preview.txt]: Ah I see, that's quite nice, a bit like telescope.

File: doc/nvim-tree.lua-float-preview.txt (hunk 3 above)
Comment (alex-courtis) [reply doc/nvim-tree.lua-float-preview.txt]: That sounds reasonable.

**Comment by alex-courtis** [comment]
I was just putting together a skeleton for you to complete. I'll leave it to you now to make it correct ;)

You have write access to my repo and can make changes / reject / merge as you need.

**Pull request merged**